v2 - Rewrite to TypeScript with generated types

lilianalillyy opened the pull request:
## Related Issues & PRs
closes PR #44
fixes issues #47, #48, #51

## Proposed Changes
 * Adds types for requests and responses generated from the Swagger schema
 * Uses Vite for build
 * Removes need for babel in favor of ts-jest in tests
 * Added `countries` as there is (to my knowledge) no endpoint to get all available countries

lilianalillyy committed:
Rewrite to TypeScript

- Adds types for requests and responses generated from the Swagger schema
- Uses Vite for build
- Removes need for babel in favor of ts-jest in tests
- Added `countries` as there is (to my knowledge) no endpoint to get all available countries

kubahasek commented:
Hey, is there any progress? Currently dealing with an error in my Remix app regarding missing declaration files. Thanks

lilianalillyy commented:
> Hey, is there any progress? Currently dealing with an error in my Remix app regarding missing declaration files. Thanks

had the same issue, you can use @lilianaa/nordigen-node for now, which is what's in this PR

kubahasek commented:
@lilianalillyy alright, will give that a shot, thanks!

kubahasek commented:
@lilianalillyy I tried to by running `npm install git+git@github.com:lilianalillyy/nordigen-node.git`, but that only resulted in this being installed, any idea why? 
<img width="322" alt="image" src="https://github.com/nordigen/nordigen-node/assets/28836407/50a03b36-9fdb-461f-9b11-1c2c21efda7f">

lilianalillyy committed:
Re-generate API types

- Enums seems to have been removed from the Swagger spec, however they were kept and are now in `src/types/enums.ts`

lilianalillyy commented:
> @lilianalillyy thank you!!

np, don't forget to update to 2.0.0-7, it's up to date with current changes

Snazzie commented:
progress? cant believe go cardless is leaving integration libraries unmaintained.
